### PR TITLE
Update TLS with code cleanup and project updates

### PR DESCRIPTION
**Pull request by NSProgrammer**
- reformatting for consistency
- xcconfig use to share settings amongst all targets, and to see comments for settings
- static functions employed (vs private methods) based on evidence of better performance

### CODE REVIEW

File: Classes/TLSLog.h [hunk 1]
```diff
@@ -47,7 +47,7 @@ NS_ASSUME_NONNULL_BEGIN
 #pragma mark Convenience Functions
 
 //! Convert the log level to a short parsable string
-FOUNDATION_EXTERN NSString *TLSLogLevelToString(TLSLogLevel level) __attribute__((const));
+FOUNDATION_EXTERN NSString *TLSLogLevelToString(TLSLogLevel level);
```
Comment (twitterkb) [inline Classes/TLSLog.h]: nitpick: why did you remove `__attribute__((const))` from this function, but not from `TLSLogChannelApplicationDefault()` just below ?  don't they both return references to `NSString * const` internally?

**Comment by twitterkb** [comment]
just the one question about the code … and it may not require a change. 

for the overall patch, it would be nice if the commit msg for this commit could contain a bit of (sanitized) detail w.r.t.

(e.g.

> - reformatting for consistency
> - xcconfig use to share settings amongst all targets, and to see comments for settings
> - static functions employed (vs private methods) based on evidence of better performance

etc.)

maybe also more info about this in the History file …

File: Classes/TLSLog.h (hunk 1 above)
Comment (NSProgrammer) [reply Classes/TLSLog.h]: So, the issue is that `TLSLogLevelToString` was updated to provide a debug string when an invalid level is provided which is dynamically created (not const) and thus if you provide an invalid level you risk crashing from compiler optimization.  It's super unlikely, but it's still a bug and removing the attribute fixes that.

**Comment by NSProgrammer** [comment]
@twitterkb I still have more passes to go through in this branch.  Once things are ready, I'll be sure the "merge" commit message is much more detailed, since that's the one that will persist most notably.
